src/transformers/modeling_utils.py

```diff
@@ -2283,37 +2283,75 @@ def tie_weights(self, missing_keys: Optional[set[str]] = None, recompute_mapping
         If `recompute_mapping=True`, it will re-check all internal submodels and their config to determine the params
         that need to be tied. This is the default when `model.tie_weights()` is called on its own, outside of
         `__init__`, and `from_pretrained`, in case the config values were changed somewhere.
+
+        Note that during `from_pretrained`, this function will work symmetrically, i.e. if the `source` is a missing
+        weight, but the `target` is present, tying will still work in reverse.
         """
         # In this case, the keys stored in `all_tied_weights_keys` are already correct
         if not recompute_mapping:
             tied_keys = self.all_tied_weights_keys
         else:
             tied_keys = self.get_expanded_tied_weights_keys(all_submodels=True)
 
-        for target_param_name, source_param_name in tied_keys.items():
-            source_param = self.get_parameter_or_buffer(source_param_name)
-            if "." in target_param_name:
-                parent_name, name = target_param_name.rsplit(".", 1)
-                parent = self.get_submodule(parent_name)
-            else:
-                name = target_param_name
-                parent = self
-            setattr(parent, name, source_param)
-            self._adjust_bias(parent, source_param)
+        tied_keys = list(tied_keys.items())
+        for i, (target_param_name, source_param_name) in enumerate(tied_keys):
+            # If both source and target are missing later, we may need to tie 2 weights so we use a list here
```

Comment:
If I understand the logic?

```suggestion
            # Usually we tie a single target to a single source, but when both are missing we may later tie
            # multiple names to a third "backup" parameter that is present in the checkpoint.
```